syntax = "proto3";

message DBRoleData {
    uint64      AccountID    = 1;
    uint64      RoleID       = 2;
    int32       CarrerID     = 3;
    int32       Level        = 4;
    int64       Exp          = 5;
    string      Name         = 6;
    int32       LangID       = 7;
    int32       VipLvl        = 8;
    int32       VipExp       = 9;
    int64       FightValue   = 10;
    int32       CityCopyID   = 11;
    repeated    int64 Action = 12;
    repeated    int64 Actime = 13;
    uint64      CreateTime   = 14;
    uint64      LogonTime    = 15;
    uint64      LogoffTime   = 16;
    uint64      GuildID      = 17;
    int32       Channel      = 18;
}

message DBBagItem {
    uint64    Guid = 1;            //格子的guid
    uint64    RoleID = 2;            //角色ID
    uint64    ItemGuid = 3;        //物品guid
    uint32    ItemID = 4;            //道具ID
    int64     Count = 5;            //个数
    int32     Bind = 6;            //是否是邦定
    int32     Status = 7;            //是否删除
}

message DBBagData {
    repeated DBBagItem ItemList = 1;
}

message DBCopyItem {
    uint64  RoleID = 1;            //角色ID
    uint32  CopyID = 2;
    int32   StarNum = 3;
    int32   BattleCnt = 4;
    int32   ResetCnt = 5;
    uint64  BattleTime = 6;
    uint64  ResetTime = 7;
}

message DBChapterItem {
    uint64  RoleID = 1;            //角色ID
    uint32  CopyType = 2;
    uint32  ChapterID = 3;
    uint32  StarAward = 4;
    uint32  SceneAward = 5;
}

message DBCopyData {
    repeated DBCopyItem CopyList = 1;
    repeated DBChapterItem ChapterList = 2;
}

message DBEquipItem {
    uint64   RoleID = 1;         //角色ID
    uint64   Guid = 2;
    int32    EquipID = 3;
    int32    StrengthLvl = 4;   //强化等级
    int32    RefineLevel = 5;   //精炼等级
    int32    StarLevel = 6;      //星级
    int32    RefineExp = 7;      //精验的经验
    int32    StarExp = 8;      //星级经验
    bool     IsUsing = 9;       //是否使用中
}

message DBEquipData {
    repeated    DBEquipItem EquipList = 1;
}

message DBGemItem {
    uint64   RoleID = 1;         //角色ID
    uint64   Guid = 2;
    int32    GemID = 3;
    int32    StrengthLvl = 4;   //强化等级
    int32    RefineLevel = 5;   //精炼等级
    int32    StarLevel = 6;      //星级
    int32    RefineExp = 7;      //精验的经验
    int32    StarExp = 8;          //星级经验
    int32    Pos = 9;        //位置
}

message DBGemData {
    repeated    DBGemItem GemList = 1;
}


message DBSkillItem {
    uint64   RoleID = 1;         //角色ID
    uint32   SkillID = 2;        //技能ID
    int32    Level = 3;          //技能等级
    int32    KeyPos = 4;         //按键位置
}

message DBSkillData {
    repeated    DBSkillItem SkillList = 1;
}


message DBPetItem {
    uint64   RoleID = 1;         //角色ID
    uint64   Guid = 2;
    int32    PetID = 3;
    int32    StrengthLvl = 4;   //强化等级
    int32    RefineLevel = 5;   //精炼等级
    int32    StarLevel = 6;      //星级
    bool     IsUsing = 7;       //是否使用中
}

message DBPetData {
    repeated    DBPetItem PetList = 1;
}

message DBMountItem {
    uint64   RoleID = 1;         //角色ID
    uint64   Guid = 2;
    int32    MountID = 3;
    int32    StrengthLvl = 4;   //强化等级
    int32    RefineLevel = 5;   //精炼等级
    int32    StarLevel = 6;      //星级
    bool     IsUsing = 7;       //是否使用中
}

message DBMountData {
    repeated    DBMountItem MountList = 1;
}

message DBPartnerItem {
    uint64   RoleID = 1;         //角色ID
    uint64   Guid = 2;
    int32    PartnerID = 3;
    int32    StrengthLvl = 4;   //强化等级
    int32    RefineLevel = 5;   //精炼等级
    int32    StarLevel = 6;      //星级
    int32    SetPos = 7;       //上阵位置
}

message DBPartnerData {
    repeated DBPartnerItem PartnerList = 1;
}

message DBTaskItem {
    uint64  RoleID = 1;            //角色ID
    uint32  TaskID = 2;
    int32   Status = 3;
    int32   Progress = 4;
}

message DBTaskData {
    repeated DBTaskItem TaskList = 1;
}

message DBMailItem {
    uint64   RoleID    = 1;   //角色ID
    uint64   Guid      = 2;   //邮件guid
    uint64   GroupID   = 3;   //群邮件guid
    uint64   Time      = 4;   //邮件时间
    uint64   SenderID  = 5;   //发送者ID
    string   Sender    = 6;   //发送者名字
    uint32   MailType  = 7;
    uint32   Status    = 8;
    string   Title     = 9;   //邮件标题
    string   Content   = 10;  //邮件内容
    bytes    Items     = 11;  //邮件物品
}

message DBMailData {
    repeated DBMailItem MailList = 1;
}

message DBActivityItem {
    uint64    RoleID = 1;            //角色ID
    uint32    ActivityID = 2;  //活动ID
    uint32    ActivityType = 3;//活动类型
    uint32    JoinTime = 4;    //参与时间
    uint32    DataLen = 5;     //数据时间长度
    bytes     Data = 6;    //活动数据
}

message DBActivityData {
    repeated DBActivityItem ActivityList = 1;
}


message DBCounterItem {
    uint64    RoleID       = 1;        //角色ID
    uint32    CounterID    = 2;        //计数器ID
    uint32    Index        = 3;        //子索引
    int64     Value        = 4;        //计数器值
    uint64    Time         = 5;        //参与时间
    uint32    Param1       = 6;        //参数1
    uint32    Param2       = 7;        //参数2
}

message DBCounterData {
    repeated DBCounterItem CounterList = 1;
}


message DBFriendItem {
    uint64    RoleID        = 1;        //角色ID
    uint64    FriendID      = 2;        //好友ID
}

message DBFriendData {
    repeated DBFriendItem FriendList = 1;
}



message DBRoleLoginAck {
    uint32              RetCode     = 1;
    uint64              RoleID      = 2;
    DBRoleData          RoleData    = 3;
    DBBagData           BagData     = 4;
    DBCopyData          CopyData    = 5;
    DBEquipData         EquipData   = 6;
    DBGemData           GemData     = 7;
    DBPetData           PetData     = 8;
    DBPartnerData       PartnerData = 9;
    DBTaskData          TaskData    = 10;
    DBMailData          MailData    = 11;
    DBMountData         MountData   = 12;
    DBActivityData      ActivityData= 13;
    DBCounterData       CounterData = 14;
    DBFriendData        FriendData  = 15;
    DBSkillData         SkillData   = 16;
}

